user-info-provider/src/main/java/de/intension/mapper/oidc/RoleBasedUserInfoProviderMapper.java

```diff
@@ -0,0 +1,135 @@
+
+import de.intension.mapper.user.UserInfoHelper;
+
+/**
+ * Provides a custom implementation based on the user property protocol mapper. This custom logic
+ * includes the checking of roles in user attributes and then adding the user property to the claim.
+ */
+public class RoleBasedUserInfoProviderMapper extends AbstractOIDCProtocolMapper
+    implements OIDCAccessTokenMapper, OIDCIDTokenMapper, UserInfoTokenMapper
+{
+
+    public static final String                        ROLES_ATTRIBUTE_NAME     = "professionalRoles";
+    public static final String                        ROLES_LABEL_NAME         = "Professional roles";
+    public static final String                        DEFAULT_ROLE             = "LEHR";
+
+    public static final String                        NEGATE_OUTPUT_NAME       = "negateOutput";
+    public static final String                        NEGATE_OUTPUT_LABEL_NAME = "Negate output";
+
+    private static final String                       CATEGORY                 = "Role based user property mapper";
+    public static final String                        PROVIDER_ID              = "roles-based-user-info-provider-mapper";
+
+    private static final List<ProviderConfigProperty> configProperties         = new ArrayList<ProviderConfigProperty>();
+    private static final UserInfoHelper               userInfoHelper           = new UserInfoHelper();
+
+    static {
+        ProviderConfigProperty userAttrProperty = new ProviderConfigProperty();
+        userAttrProperty.setName(ProtocolMapperUtils.USER_ATTRIBUTE);
+        userAttrProperty.setLabel(ProtocolMapperUtils.USER_MODEL_PROPERTY_LABEL);
+        userAttrProperty.setType(ProviderConfigProperty.STRING_TYPE);
+        userAttrProperty.setHelpText(ProtocolMapperUtils.USER_MODEL_PROPERTY_HELP_TEXT);
+        configProperties.add(userAttrProperty);
+
+        ProviderConfigProperty rolesProperty = new ProviderConfigProperty();
+        rolesProperty.setName(ROLES_ATTRIBUTE_NAME);
+        rolesProperty.setLabel(ROLES_LABEL_NAME);
+        rolesProperty.setDefaultValue(DEFAULT_ROLE);
+        rolesProperty.setType(ProviderConfigProperty.STRING_TYPE);
+        rolesProperty.setHelpText("Includes comma separated list of roles which needs to be checked in the user attributes");
+        configProperties.add(rolesProperty);
+
+        ProviderConfigProperty negateOutput = new ProviderConfigProperty();
+        negateOutput.setName(NEGATE_OUTPUT_NAME);
+        negateOutput.setLabel(NEGATE_OUTPUT_LABEL_NAME);
+        negateOutput.setDefaultValue(false);
+        negateOutput.setType(ProviderConfigProperty.BOOLEAN_TYPE);
+        negateOutput
+            .setHelpText("Apply a NOT to the check result of roles. When this is true, then the condition will evaluate to true just if user does NOT have the specified role present in the user attributes. When this is false, the condition will evaluate to true just if user has the specified role");
```

Comment:
This would be nice in the README too!

What does it **mean** when the mapper "evaluates to true/false"? that part is missing.

My guess:
Token gets the claim of the role when:
1. Negate output is false. User has role in 1 of the sanis attributes.
2. Negate output is true. User doesn't have role in 1 of the sanis attributes.

Reply:
Refer decision matrix in https://intension-de.atlassian.net/browse/FWUV-240

Reply:
Also refer to the updated documentation